test/Test/Data/Mutable/Array.hs

```diff
@@ -6,31 +6,13 @@
 -- |
 -- Tests for mutable arrays.
 --
--- The API is basically 4 mutators or constructors: alloc, write, resize,
--- resizeSeed and 2 accessors: length, read.
+-- See the testing framework explained in Test.Data.Mutable.Set.
 --
--- We test arrays behave properly with properties that axiomatically define
--- arrays.  Basically, using these properties, the behavior of any use of
--- arrays should be fully specified without having to look at the
--- implementation.
+-- The combiniation of axioms and homormorphisms provided fully specify
```

Comment:
I find this sentence

> fully specify the behavior of arrays.

very ambitious. I'm sure we can find some operations outside `read` and `write` that we don't "fully" specify, I'm thinking about things like `freeze`. And whether something specifies it "fully" is also open, because then we should also test the out of bounds conditions etc. I'd honestly just remove the sentence.

Reply:
I completely disagree. It's the whole point of #93. 

Right now the sentence means "fully specify" minus the remarks of things we skip. It's clear that that is true.

If you have a reference implementation and define homomorphisms for every constructor/modifer/accessor `f` of the form `f (fromList l) = fromList (f' l)` or `f (fromList l) = f' l`, then you can always move computations inside the `fromList` and have axioms that define everything.

Reply:
The explanation in Test.Data.Mutable.Set in #263 covers this with an example proof.

Reply:
“fully specifies” is a strong word. First, this is only functional specification. None of these properties specify anything about, say, the cost semantics of arrays. Second, of course, it does mean that we have a semantics for every single operation.

It's a rather good idea to tone it down. Something like “functionally specifies each operation”, maybe?

Or really, nothing at all, since you already point to the Set module which has less terse explanation.